src/main/java/seedu/ccacommander/model/CommandHistory.java

```diff
@@ -0,0 +1,90 @@
+package seedu.ccacommander.model;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * The facilitator of commands and functions related to recalling previously entered commands.
+ */
+public class CommandHistory {
+    private List<String> commandHistoryList;
+    private int currentCommandPointer;
+
+    /**
+     * Constructs a new instance of CommandHistory.
+     */
+    public CommandHistory() {
+        this.commandHistoryList = new ArrayList<>();
+        this.currentCommandPointer = 0;
+    }
+
+    /**
+     * Adds the command into {@code commandHistoryList} and sets the pointer to the new command only if the command
+     * is not the latest command added.
+     *
+     * @param command
+     */
+    public void addCommand(String command) {
+        requireNonNull(command);
+        if (this.commandHistoryList.size() == 0
+                || !this.commandHistoryList.get(this.commandHistoryList.size() - 1).equals(command)) {
```

Comment:
Is there a possible way to abstract the 2nd condition check to another boolean method? This is to prevent a long if condition.

Reply:
Sure, good idea